dask/array/core.py

```diff
@@ -2812,6 +2812,12 @@ def insert(arr, obj, values, axis):
 def broadcast_to(x, shape):
     shape = tuple(shape)
 
+    # Promote Python scalars and NumPy arrays
+    if not hasattr(x, "shape"):
+        x = np.array(x)
+    if not hasattr(x, "chunks"):
+        x = from_array(x, chunks=x.shape)
```

Comment:
This would let `h5py.Dataset` objects through (or anything else with `shape` and `chunks`). A better fix might be to replace this block with `x = asarray(x)` (using `dask.array.core.asarray`).